gcloud-java-compute/src/test/java/com/google/gcloud/compute/it/ITComputeTest.java

```diff
@@ -637,4 +642,217 @@ public void testListZoneOperationsWithFilter() {
+    assertTrue(remoteAddress.addressId() instanceof RegionAddressId);
+    assertEquals(REGION, remoteAddress.<RegionAddressId>addressId().region());
+    assertEquals(addressId.address(), remoteAddress.addressId().address());
+    assertNull(remoteAddress.address());
+    assertNull(remoteAddress.creationTimestamp());
+    assertNull(remoteAddress.id());
+    operation = remoteAddress.delete();
+    while (!operation.isDone()) {
+      Thread.sleep(1000L);
+    }
+    assertNull(compute.get(addressId));
+  }
+
+  @Test
+  public void testListRegionAddresses() throws InterruptedException {
+    String prefix = BASE_RESOURCE_NAME + "list-region-address";
+    String[] addressNames = {prefix + "1", prefix + "2"};
+    AddressId firstAddressId = RegionAddressId.of(REGION, addressNames[0]);
+    AddressId secondAddressId = RegionAddressId.of(REGION, addressNames[1]);
+    Operation firstOperation = compute.create(AddressInfo.of(firstAddressId));
+    Operation secondOperation = compute.create(AddressInfo.of(secondAddressId));
+    while (!firstOperation.isDone()) {
+      Thread.sleep(1000L);
+    }
+    while (!secondOperation.isDone()) {
+      Thread.sleep(1000L);
+    }
+    Set<String> addressSet = ImmutableSet.copyOf(addressNames);
+    // test list
+    Compute.AddressFilter filter =
+        Compute.AddressFilter.equals(Compute.AddressField.NAME, prefix + "\\d");
+    Page<Address> addressPage =
+        compute.listRegionAddresses(REGION, Compute.AddressListOption.filter(filter));
+    Iterator<Address> addressIterator = addressPage.iterateAll();
+    int count = 0;
+    while (addressIterator.hasNext()) {
+      Address address = addressIterator.next();
+      assertNotNull(address.addressId());
+      assertTrue(address.addressId() instanceof RegionAddressId);
+      assertEquals(REGION, address.<RegionAddressId>addressId().region());
+      assertTrue(addressSet.contains(address.addressId().address()));
+      assertNotNull(address.address());
+      assertNotNull(address.creationTimestamp());
+      assertNotNull(address.id());
+      count++;
+    }
+    assertEquals(2, count);
+    // test list with selected fields
+    count = 0;
+    addressPage = compute.listRegionAddresses(REGION, Compute.AddressListOption.filter(filter),
+        Compute.AddressListOption.fields(Compute.AddressField.ADDRESS));
+    addressIterator = addressPage.iterateAll();
+    while (addressIterator.hasNext()) {
+      Address address = addressIterator.next();
+      assertTrue(address.addressId() instanceof RegionAddressId);
+      assertEquals(REGION, address.<RegionAddressId>addressId().region());
+      assertTrue(addressSet.contains(address.addressId().address()));
+      assertNotNull(address.address());
+      assertNull(address.creationTimestamp());
+      assertNull(address.id());
+      assertNull(address.status());
+      assertNull(address.usage());
+      count++;
+    }
+    assertEquals(2, count);
+    compute.delete(firstAddressId);
```

Comment:
Here and in other test list methods here: should we wait until the addresses are deleted?  (Could the created addresses interfere with the testAggregatedListAddresses method?)